Makefile.am

```diff
@@ -93,6 +94,21 @@ mcs-do-compiler-tests:
 bootstrap-world: compiler-tests
 	$(MAKE) install
 
+install:
+	echo $(SUBDIRS)
+	for mydir in $(SUBDIRS) ; do \
+		do_install=1 ; \
+		for noinstdir in $(noinst_SUBDIRS) ; do \
+			if [ "x$${noinstdir}" = "x$${mydir}" ] ; then \
+				do_install=0 ; \
+				break ; \
+			fi \
+		done ; \
+		if [ "$${do_install}" = "1" ] ; then \
+			(cd $${mydir} && ${MAKE} install) \
+		fi \
+	done
```

Comment:
There are `install-local` targets, right?  If there was one in this Makefile.am, it would still run, right? (I assume automake adds `install: install-local`) You don't have to call it explicitly?

Reply:
We're talking about BTLS in this case, yes?

`mono/btls/Makefile.am` contains a `install-exec-local` target.

The generated `mono/btls/Makefile` generates an `install-exec-am` target, which depends on `install-exec-local`. It generates an `install-exec` target which depends on `install-exec-am`. It generates an `install-am` target which manually calls (but technically does not depend on) `@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am`. It generates an `install` target which depends on `install-am`. So, long-winded answer: yes Automake deals with this for you.